cocaine/proxy/proxy.py

```diff
@@ -114,6 +116,49 @@
 
 X_COCAINE_HTTP_PROTO_VERSION = "X-Cocaine-HTTP-Proto-Version"
 
+class BodyProcessor(object):
+
+    def __init__(self, request, name, code, headers, init=''):
+        self.request = request
+        self.name = name
+        self.code = code
+        self.headers = headers
+        self.message = init
+
+    def __call__(self, part):
+        pass
+
+    def finish(self):
+        pass
+
+class ChunkedBodyProcessor(BodyProcessor):
+
+    def __init__(self, request, name, code, headers, init=''):
+        super(ChunkedBodyProcessor, self).__init__(
+            request, name, code, headers, init)
+
+        self.headers['X-Cocaine-Application'] = self.name
+        self.headers['Transfer-Encoding'] = 'chunked'
```

Comment:
I don't know if the tornado reorders headers (likely not), but it's more intuitive to write common and well-known headers first.